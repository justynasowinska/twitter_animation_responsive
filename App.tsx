import React, { useState } from 'react';
import { StyleSheet, View, Dimensions, LayoutChangeEvent } from 'react-native';

import { MainScreen } from './src/MainScreen';
import { ScreenSizeContext } from './src/context/ScreenSizeContext';
import { colors } from './src/utils/colors';

export default function App() {
  const [screenHeight, setScreenHeight] = useState(Dimensions.get('window').height);

  const onLayoutHandle = (e: LayoutChangeEvent) => {
    setScreenHeight(e.nativeEvent.layout.height);
  };

  return (
    <View style={styles.container} onLayout={onLayoutHandle}>
      <ScreenSizeContext.Provider value={{ height: screenHeight }}>
        <MainScreen />
      </ScreenSizeContext.Provider>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.appBackgroundColor
  }
});
