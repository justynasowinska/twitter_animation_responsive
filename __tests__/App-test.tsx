import { shallow } from 'enzyme';
import React from 'react';
import App from '../App';

it('renders correctly', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
});
