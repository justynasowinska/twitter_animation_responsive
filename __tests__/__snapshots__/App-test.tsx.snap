// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "height": 1334,
      }
    }
  >
    <MainScreen />
  </ContextProvider>
</View>
`;
